centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php

```diff
@@ -134,6 +135,48 @@ public function exists(int $hostId): bool
         return (bool) $statement->fetchColumn();
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function existsByAccessGroups(int $hostId, array $accessGroups): bool
+    {
+        if (empty($accessGroups)) {
+            $this->debug('Access groups array empty');
+
+            return false;
+        }
+
+        $accessGroupIds = array_map(
+            fn ($accessGroup) => $accessGroup->getId(),
```

Comment:
The higher the phpstan level is, the lower it relies on inference.
Please be the most explicit possible, even if obvious.
```suggestion
            static fn (AccessGroup $accessGroup): int => $accessGroup->getId(),
```